src/Umbraco.Web/Models/Mapping/MemberModelMapper.cs

```diff
@@ -287,20 +287,21 @@ internal static ContentPropertyDisplay GetLoginProperty(IMemberService memberSer
 
         internal static IDictionary<string, bool> GetMemberGroupValue(string username)
         {
-            var result = new Dictionary<string, bool>();
-            foreach (var role in Roles.GetAllRoles().Distinct())
-            {
+            var userRoles = username.IsNullOrWhiteSpace() ? null : Roles.GetRolesForUser(username);
+
+            // create a dictionary of all roles (except internal roles) + "false"
+            var result = Roles.GetAllRoles().Distinct()
                 // if a role starts with __umbracoRole we won't show it as it's an internal role used for public access
-                if (role.StartsWith(Constants.Conventions.Member.InternalRolePrefix) == false)
-                {
-                    result.Add(role, false);
-                    if (username.IsNullOrWhiteSpace()) continue;
-                    if (Roles.IsUserInRole(username, role))
-                    {
-                        result[role] = true;
-                    }
-                }
-            }
+                .Where(x => x.StartsWith(Constants.Conventions.Member.InternalRolePrefix) == false)
+                .ToDictionary(x => x, x => false);

```

Comment:
This is all good with this change of comparing the two group of items as opposed.
No problem, but the nit-picker & Mr Verbose hacker that I am I prefer LINQ params not to be x and use something more descriptive such as role perhaps, but again that's just me :)